tests/trainer/test_trainer.py

```diff
@@ -996,33 +1000,60 @@ def test_can_resume_training(self):
             trainer.train(resume_from_checkpoint=True)
         self.assertTrue("No valid checkpoint found in output directory" in str(context.exception))
 
-    @require_torch_non_multi_gpu
     def test_resume_training_with_randomness(self):
-        # This test will fail flakily for more than 1 GPUs since the result will be slightly more different
-        # TODO: investigate why it fails for 2 GPUs?
+        # For more than 1 GPUs, since the randomness is introduced in the model and with DataParallel (which is used
+        # in this test for more than 2 GPUs), the calls to the torch RNG will happen in a random order (sometimes
+        # GPU 0 will call first and sometimes GPU 1).
+        random_torch = torch.cuda.is_available() and torch.cuda.device_count() >= 1
```

Comment:
Sorry, just a question regarding this line. AFAICT `random_torch` would only be `True` if at least one GPU is available. But this would mean this test case will not cover `torch` randomness when using the CPU. The unit test before this commit however _did_ test randomness on the CPU. Is this change intended?

Reply:
Good catch! I'll fix this :-)